src/main/java/randoop/condition/ConditionMethodCreator.java

```diff
@@ -0,0 +1,204 @@
+package randoop.condition;
+
+import java.lang.reflect.Method;
+import java.util.List;
+import javax.tools.Diagnostic;
+import javax.tools.JavaFileObject;
+import randoop.Globals;
+import randoop.compile.SequenceCompiler;
+import randoop.compile.SequenceCompilerException;
+import randoop.output.NameGenerator;
+import randoop.reflection.RawSignature;
+
+/** Defines the factory method for creating condition methods. */
+public class ConditionMethodCreator {
+
+  /** The basename for the condition class name. */
+  private static final String CONDITION_CLASS_BASENAME = "RandoopConditionClass";
+
+  /** The name of the condition method. */
+  private static final String CONDITION_METHOD_NAME = "test";
+
+  /** The name generator used to generate class names. */
+  private static final NameGenerator nameGenerator = new NameGenerator(CONDITION_CLASS_BASENAME);
+
+  /**
+   * Creates the {@code java.lang.reflect.Method} to test the condition in the condition code.
+   *
+   * <p>Generates the Java source for a class with the method and then
+   *
+   * @param signature the signature string for the condition method
+   * @param parameterDeclaration the parameter declaration string, includes parentheses
```

Comment:
What is the relationship of `parameterDeclaration` to `signature`?  How is this different than the formal parameters in `signature`?